File: typescript/src/theme.ts
export interface Theme {
  question: string;
  success: string;
  pointer: string;
  highlight: string;
  selected: string;
  answer: string;
  error: string;
  muted: string;
  symQuestion: string;
  symSuccess: string;
  symPointer: string;
  symChecked: string;
  symUnchecked: string;
}

export const defaultTheme: Theme = {
  question: "#9fa4e3",
  success: "#62bfa1",
  pointer: "#9c99ec",
  highlight: "#90bbe9",
  selected: "#59bca4",
  answer: "#9db9dd",
  error: "#d77780",
  muted: "#84858f",
  symQuestion: "?",
  symSuccess: "✓",
  symPointer: "❯",
  symChecked: "◉",
  symUnchecked: "◯",
};

let currentTheme: Theme = { ...defaultTheme };

export function setTheme(theme: Partial<Theme>): void {
  currentTheme = { ...currentTheme, ...theme };
}

export function getTheme(): Theme {
  return currentTheme;
}

export const RESET = "\x1b[0m";
export const BOLD = "\x1b[1m";

export function ansi(hexColor: string): string {
  const h = hexColor.replace("#", "");
  const r = parseInt(h.substring(0, 2), 16);
  const g = parseInt(h.substring(2, 4), 16);
  const b = parseInt(h.substring(4, 6), 16);
  return `\x1b[38;2;${r};${g};${b}m`;
}

File: rust/src/theme.rs
pub struct Theme {
    pub question: &'static str,
    pub success: &'static str,
    pub highlight: &'static str,
    pub selected: &'static str,
    pub answer: &'static str,
    pub error: &'static str,
    pub muted: &'static str,
    pub sym_question: &'static str,
    pub sym_success: &'static str,
    pub sym_pointer: &'static str,
    pub sym_checked: &'static str,
    pub sym_unchecked: &'static str,
}

pub const DEFAULT_THEME: Theme = Theme {
    question: "#9fa4e3",
    success: "#62bfa1",
    highlight: "#90bbe9",
    selected: "#59bca4",
    answer: "#9db9dd",
    error: "#d77780",
    muted: "#84858f",
    sym_question: "?",
    sym_success: "✓",
    sym_pointer: "❯",
    sym_checked: "◉",
    sym_unchecked: "◯",
};

pub const RESET: &str = "\x1b[0m";
pub const BOLD: &str = "\x1b[1m";

pub fn ansi_color(hex: &str) -> String {
    let h = hex.trim_start_matches('#');
    let r = u8::from_str_radix(&h[0..2], 16).unwrap_or(0);
    let g = u8::from_str_radix(&h[2..4], 16).unwrap_or(0);
    let b = u8::from_str_radix(&h[4..6], 16).unwrap_or(0);
    format!("\x1b[38;2;{r};{g};{b}m")
}
